src/app/lightning/delete-ln-payments.ts

```diff
@@ -0,0 +1,98 @@
+import { UnknownLightningServiceError } from "@domain/bitcoin/lightning"
+import { LndService } from "@services/lnd"
+import { LnPaymentsRepository } from "@services/mongoose"
+import {
+  addAttributesToCurrentSpan,
+  asyncRunInSpan,
+  SemanticAttributes,
+} from "@services/tracing"
+
+export const deleteLnPaymentsBefore = async (
+  timestamp: Date,
+): Promise<true | ApplicationError> => {
+  const paymentHashesBefore = await listAllPaymentsBefore(timestamp)
+
+  for await (const paymentHash of paymentHashesBefore) {
+    if (paymentHash instanceof Error) return paymentHash
+    await checkAndDeletePaymentForHash(paymentHash)
+  }
+
+  return true
+}
+
+const checkAndDeletePaymentForHash = async ({
+  paymentHash,
+  pubkey,
+}: {
+  paymentHash: PaymentHash
+  pubkey: Pubkey
+}): Promise<boolean | ApplicationError> =>
+  asyncRunInSpan(
```

Comment:
if we are using Lightning in cron then we are wrapping this method twice. I think you can remove this wrapper and use the automatic wrapper from app index

Reply:
The extra wrapper here is useful for getting more granular with certain attributes like `deleted`

- Trace without extra wrapper: https://ui.honeycomb.io/galoy/datasets/arvin-dev/result/v631Ah4Exxj/trace/yF7iqYvUk19?span=a36a1938aa52cf30
- Trace with extra wrapper: https://ui.honeycomb.io/galoy/datasets/arvin-dev/result/o9hyBy1s1pN/trace/JGYUFWQi3Uu?span=2abeaa8603d0ec78